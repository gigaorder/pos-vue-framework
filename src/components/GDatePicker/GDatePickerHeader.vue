<template>
  <div :class="datePickerHeaderClasses">
    <div :class="datePickerHeaderValueClasses">
      <transition :name="headerTransitionName">
        <div :key="value" :class="headerData.class" :style="headerData.style">
          <button type="button" v-on:click="onHeaderClicked">
            <template v-if="defaultSlotIsAvailable">
              <slot></slot>
            </template>
            <template v-else>
              {{ headerFormatter(value) }}
            </template>
          </button>
        </div>
      </transition>
    </div>
    <button :disabled="prevBtnDisabled" class="g-date-picker-header__prev-button" v-on:click="prevClick">
<!--      <i class="material-icons">mdi-chevron-left</i>-->
    </button>
    <button :disabled="nextBtnDisabled" class="g-date-picker-header__next-button" v-on:click="nextClick">
<!--      <i class="material-icons">mdi-chevron-right</i>-->
    </button>
  </div>
</template>

<script>
  import { computed, reactive, watch } from '@vue/composition-api'
  import { monthChange, createNativeLocaleFormatter, TRANSITION_NAMES } from './utils'
  import { setTextColor } from '../../mixins/colorable'
  import GButton from '../GButton/GButton'

  // clss
  const DATE_PICKER_HEADER_CLS = {
    HEADER: 'g-date-picker-header',
    HEADER_DISABLED: 'g-date-picker-header--disabled',
    HEADER_VALUE: 'g-date-picker-header__value',
    HEADER_VALUE_DISABLED: 'g-date-picker-header__value--disabled'
  }

  const NAV = {
    PREV: -1,
    NEXT: 1
  }

  export const EVENT_NAMES = {
    TOGGLE: 'toggle',
    INPUT: 'input'
  }

  export default {
    name: 'GDatePickerHeader',
    components: { GButton },
    props: {
      disabled: Boolean,
      readonly: Boolean,
      min: String,
      max: String,
      value: {
        type: [Number, String],
        required: true,
      },
      format: Function,
    },
    setup(props, context) {
      // header transition
      const state = reactive({ isReversing: false })
      const headerTransitionName = computed(() => (state.isReversing) ? TRANSITION_NAMES.REVERSE_TAB : TRANSITION_NAMES.TAB)
      watch(() => props.value, (newVal, oldVal) => state.isReversing = newVal < oldVal, { lazy: true, flush: 'pre' })
      // header value
      const headerFormatter = computed(() => {
        if (props.format) {
          return props.format
        } else if (String(props.value).split('-')[1]) {
          return createNativeLocaleFormatter(undefined, { month: 'long', year: 'numeric', timeZone: 'UTC' }, { length: 7 })
        } else {
          return createNativeLocaleFormatter(undefined, { year: 'numeric', timeZone: 'UTC' }, { length: 4 })
        }
      })
      const datePickerHeaderClasses = computed(() => {
        return {
          [DATE_PICKER_HEADER_CLS.HEADER]: true,
          [DATE_PICKER_HEADER_CLS.HEADER_DISABLED]: props.disabled
        }
      })
      const datePickerHeaderValueClasses = computed(() => {
        return {
          [DATE_PICKER_HEADER_CLS.HEADER_VALUE]: true,
          [DATE_PICKER_HEADER_CLS.HEADER_VALUE_DISABLED]: props.disabled
        }
      })
      const headerData = computed(() => setTextColor(props.color, {}))
      const defaultSlotIsAvailable = context.slots.default !== undefined
      // header event handler
      const onHeaderClicked = () => context.emit(EVENT_NAMES.TOGGLE)


      // Prev button
      const prevBtnDisabled = computed(() => props.disabled || (props.min && calculateChange(NAV.PREV) < props.min))
      const prevClick = () => context.emit(EVENT_NAMES.INPUT, calculateChange(NAV.PREV))

      // Next button
      const nextBtnDisabled = computed(() => props.disabled || (props.max && calculateChange(NAV.NEXT) < props.max))
      const nextClick = () => context.emit(EVENT_NAMES.INPUT, calculateChange(NAV.NEXT))

      //
      function calculateChange(sign/*: number*/) {
        const [year, month] = String(props.value).split('-').map(Number)
        if (month == null) {
          return `${year + sign}`
        } else {
          return monthChange(props.value, sign)
        }
      }


      return {
        datePickerHeaderClasses,
        datePickerHeaderValueClasses,
        headerData,
        headerTransitionName,
        defaultSlotIsAvailable,
        onHeaderClicked,
        headerFormatter,
        //
        prevBtnDisabled,
        prevClick,
        //
        nextBtnDisabled,
        nextClick,

      }
    }
  }
</script>
<style scoped>
</style>
