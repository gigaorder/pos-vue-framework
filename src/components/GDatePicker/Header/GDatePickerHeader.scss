@import "../var";

.g-date-picker-header {
  padding: 4px 8px;
  align-items: center;
  display: flex;
  justify-content: space-between;
  position: relative ;

  .g-btn {
    margin: 0;
    z-index: auto;
  }

  .g-icon {
    cursor: pointer;
    user-select: none;
  }

  &--disabled {
    pointer-events: none;
  }

  &__value {
    flex: 1;
    position: relative;
    overflow: hidden;
    color: rgb(83, 83, 83);

    div {
      transition: $primary-transition;
      width: 100%;
    }

    button {
      cursor: pointer;
      font-weight: bold;
      outline: none;
      padding: 0.5rem;
      transition: $primary-transition;
      background-color: transparent;
      border-style: none;
      color: inherit;
    }
  }

  %nav_button {
    height: 24px;
    width: 24px;
    border: none;
    background-color: transparent;
  }

  &__prev-button {
    @extend %nav_button;
    margin-right: 24px;

    &::before {
      content: '\F141';
      font-family: "Material Design Icons";
    }
  }

  &__next-button {
    @extend %nav_button;

    &::after {
      content: '\F142';
      font-family: "Material Design Icons";
    }
  }
}


