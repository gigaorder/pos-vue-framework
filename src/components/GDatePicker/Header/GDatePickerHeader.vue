<template>
  <div :class="datePickerHeaderClasses">
    <div :class="datePickerHeaderValueClasses">
      <transition :name="headerTransitionName">
        <div :key="value" :class="headerData.class" :style="headerData.style">
          <button type="button" @click="onHeader">
            <template v-if="defaultSlotIsAvailable">
              <slot></slot>
            </template>
            <template v-else>
              {{ headerFormatter(value) }}
            </template>
          </button>
        </div>
      </transition>
    </div>
    <button :disabled="!canGoPrev" class="g-date-picker-header__prev-button" @click="onPrev"></button>
    <button :disabled="!canGoNext" class="g-date-picker-header__next-button" @click="onNext"></button>
  </div>
</template>

<script>
  import { computed, reactive, watch } from '@vue/composition-api'
  import { TRANSITION_NAMES } from '../utils'
  import { setTextColor } from '../../../mixins/colorable'
  import { getHeaderEvents, getHeaderFormatter, getNavigationState  } from './GDatePickerHeaderUtil';

  // css class
  const CSS_CLS = {
    HEADER: 'g-date-picker-header',
    HEADER_DISABLED: 'g-date-picker-header--disabled',
    HEADER_VALUE: 'g-date-picker-header__value',
    HEADER_VALUE_DISABLED: 'g-date-picker-header__value--disabled'
  }

  export default {
    name: 'GDatePickerHeader',
    props: {
      disabled: Boolean,
      readonly: Boolean,
      min: String,
      max: String,
      value: {
        type: [Number, String],
        required: true,
      },
      format: Function,
    },
    setup(props, context) {
      // header transition
      const state = reactive({ isReversing: false })

      const headerTransitionName = computed(() => (state.isReversing) ? TRANSITION_NAMES.REVERSE_TAB : TRANSITION_NAMES.TAB)
      watch(() => props.value, (newVal, oldVal) => state.isReversing = newVal < oldVal, { lazy: true, flush: 'pre' })

      // header content
      const headerFormatter = getHeaderFormatter(props)
      const datePickerHeaderClasses = computed(() => {
        return {
          [CSS_CLS.HEADER]: true,
          [CSS_CLS.HEADER_DISABLED]: props.disabled
        }
      })
      const datePickerHeaderValueClasses = computed(() => {
        return {
          [CSS_CLS.HEADER_VALUE]: true,
          [CSS_CLS.HEADER_VALUE_DISABLED]: props.disabled
        }
      })
      const headerData = computed(() => setTextColor(props.color, {}))
      const defaultSlotIsAvailable = context.slots.default !== undefined


      // navigation
      const { canGoPrev, canGoNext } = getNavigationState(props)
      const { onPrev, onNext, onHeader } = getHeaderEvents(props, context)

      return {
        datePickerHeaderClasses,
        datePickerHeaderValueClasses,
        headerData,
        headerTransitionName,
        defaultSlotIsAvailable,
        onHeader,
        headerFormatter,
        //
        canGoPrev,
        onPrev,
        //
        canGoNext,
        onNext,
      }
    }
  }
</script>
<style scoped>
</style>
