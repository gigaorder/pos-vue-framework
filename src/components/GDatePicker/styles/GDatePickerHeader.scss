// @import '../../styles/styles.sass'

//+theme(g-date-picker-header) using ($material) {
//  .g-date-picker-header__value:not(.g-date-picker-header__value--disabled) {
//    button:not(:hover):not(:focus) {
//      color: map-deep-get($material, 'text', 'primary')
//    }
//  }
//
//  .v-date-picker-header__value--disabled {
//    button {
//      color: map-deep-get($material, 'text', 'disabled')
//    }
//  }
//}

@import "var";

.g-date-picker-header {
  padding: 4px 8px;
  align-items: center;
  display: flex;
  justify-content: space-between;
  position: relative ;

  .g-btn {
    margin: 0;
    z-index: auto;
  }

  .g-icon {
    cursor: pointer;
    user-select: none;
  }

  &--disabled {
    pointer-events: none;
  }

  &__value {
    flex: 1;
    position: relative;
    overflow: hidden;

    div {
      transition: $primary-transition;
      width: 100%;
    }

    button {
      cursor: pointer;
      font-weight: bold;
      outline: none;
      padding: 0.5rem;
      transition: $primary-transition;
      background-color: transparent;
      border-style: none;
      color: inherit;
    }
  }

  %nav_button {
    height: 24px;
    width: 24px;
    border: none;
    background-color: transparent;
  }

  &__prev-button {
    @extend %nav_button;
    margin-right: 24px;
  }

  &__next-button {
    @extend %nav_button;

  }
}


