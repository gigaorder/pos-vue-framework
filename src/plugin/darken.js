import {getHslColor} from "../utils/colors";

;(function (window) {
  'use strict';

  const Darken = window.Darken || {};

  const Effect = {
    //click
    elements: [],
    // Effect delay
    duration: 1500,

    show: function (e, element) {

      // Disable right click
      if (e.button === 2) {
        return false;
      }

      const el = element || this;
      Effect.elements.push(el)

      const background = el.style.backgroundColor,
          border = el.style.borderColor,
          shadow = el.style.boxShadow

      let {h, s, l, a} = getHslColor(background)
      if (h === 0 && s === 0) {
        let hsla
        if (border)
          hsla = getHslColor(border)
        else
          hsla = {h: 0, s: 0, l: 0, a: 0}
        h = hsla.h
        s = hsla.s
        l = hsla.l
        a = hsla.a
      }
      el.style.background = `hsla(${h}, ${s - 5}%, ${l - 15}%, ${a})`
      el.style.borderColor = `hsla(${h + 2}, ${s - 10}%, ${l - 15}%, ${a})`
      el.style.boxShadow = `1px 0px 5px hsla(${h + 2}, ${s - 10}%, ${l - 15}%, ${a})`;

      setTimeout(() => {
        el.style.backgroundColor = background
        el.style.borderColor = border
        el.style.boxShadow = shadow
        const index = Effect.elements.findIndex(element => element === el)
        Effect.elements.splice(index, 1)
      }, Effect.duration)
    },

  };

  const TouchHandler = {
    touches: 0,
    allowEvent: function (e) {
      let allow = true;

      if (e.type === 'touchstart') {
        TouchHandler.touches += 1;
      } else if (e.type === 'touchend' || e.type === 'touchcancel') {
        setTimeout(function () {
          if (TouchHandler.touches > 0) {
            TouchHandler.touches -= 1;
          }
        }, Effect.duration);
      } else if (e.type === 'mousedown' && TouchHandler.touches > 0) {
        allow = false;
      }

      return allow;
    },
    touchup: function (e) {
      TouchHandler.allowEvent(e);
    }
  };

  function getWavesEffectElement(e) {
    if (TouchHandler.allowEvent(e) === false) {
      return null;
    }

    let element = null;
    let target = e.target || e.srcElement;

    while (target.parentNode !== null) {
      if (!(target instanceof SVGElement) && target.className.indexOf('darken-effect') !== -1) {
        element = target;
        break;
      }
      target = target.parentNode;
    }
    return element;
  }

  function showEffect(e) {
    const element = getWavesEffectElement(e);

    if (element !== null && Effect.elements.findIndex(el => el === element) === -1) {
      Effect.show(e, element);
    }
  }

  Darken.displayEffect = function (options) {
    options = options || {};

    if ('duration' in options) {
      Effect.duration = options.duration;
    }


    if ('ontouchstart' in window) {
      document.body.addEventListener('touchstart', showEffect, false);
    }

    document.body.addEventListener('mousedown', showEffect, false);
  };

  Darken.attach = function (element) {
    if ('ontouchstart' in window) {
      element.addEventListener('touchstart', showEffect, false);
    }

    element.addEventListener('mousedown', showEffect, false);
  };

  window.Darken = Darken;

  document.addEventListener('DOMContentLoaded', function () {
    Darken.displayEffect();
  }, false);

})(window);
