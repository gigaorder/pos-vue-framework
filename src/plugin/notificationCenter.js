import Vue from 'vue';
import GNotification from '../components/GNotification/GNotification';

export default {
  install(Vue, options) {
    const notificationCenter = new Vue({
      ...GNotification,
      propsData: options ? parseOptions(options) : {}
    })
    notificationCenter.$mount(document.body.appendChild(document.createElement('div')));

    Vue.prototype.$notificationCenter = notificationCenter;
  }
}

function parseOptions(options) {
  const propsData = {}

  if(options.position) {
    const position = options.position.split('-');
    for (let i = 0; i < position.length; i++) {
      propsData[position[i]] = true
    }
  }

  if (options.width) propsData['width'] = options.width

  return propsData;
}