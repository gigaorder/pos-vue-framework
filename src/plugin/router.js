import Vue from 'vue';
import VueRouter from 'vue-router';
import KeyboardDemo from '@/view/KeyboardDemo';
import Groupable from '@/view/Groupable';
import MenuDemo from '@/view/MenuDemo';
import SliderDemo from '../view/SliderDemo';
import OverlayDemo from "../view/OverlayDemo";
import SelectIonControl from '../view/SelectIonControl';
import DialogDemo from '../view/DialogDemo';
import PulseDemo from '../view/PulseDemo';
import ToolTipDemo from '../view/ToolTipDemo';
import InputDemo from "../view/InputDemo";
import TransitionDemo from '../view/TransitionDemo';
import ListDemo from '../view/ListDemo';
import List2Demo from "../view/List2Demo";
import DndDialogDemo from '../view/DndDialogDemo';
import GChipGroupDemo from '../view/GChipGroupDemo';
import DatePicker from '../view/DatePickerDemo';
import ExpansionPanelDemo from '../view/ExpansionPanelDemo';
import Textarea from "../view/TextareaDemo";
import LoginView from '../view/LoginView';

const routes = [
  {path: '/login', name: 'Login', component: LoginView},
];

const router = new VueRouter({
  mode: 'history',
  base: '/',
  routes
});

export default router
