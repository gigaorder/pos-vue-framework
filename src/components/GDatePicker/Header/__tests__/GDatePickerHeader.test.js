import { createLocalVue, mount } from '@vue/test-utils'
import plugin from '@vue/composition-api'
import GDatePickerHeader from '../GDatePickerHeader';

describe('GDatePickerHeader.js', () => {
  let mountFunction

  const localVue = createLocalVue()
  localVue.use(plugin)

  beforeEach(() => {
    mountFunction = (options) => {
      return mount(GDatePickerHeader, {
        localVue,
        ...options
      })
    }
  })

  it('should render component and match snapshot', () => {
    const wrapper = mountFunction({
      propsData: {
        value: '2005-11',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render disabled component and match snapshot', () => {
    const wrapper = mountFunction({
      propsData: {
        value: '2005-11',
        disabled: true,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render readonly component and match snapshot', () => {
    const wrapper = mountFunction({
      propsData: {
        value: '2005-11',
        readonly: true,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with year value and match snapshot', () => {
    const wrapper = mountFunction({
      propsData: {
        value: '2005',
      },
    })

    expect(wrapper.findAll('.g-date-picker-header__value div').at(0).element.textContent.trim()).toBe('2005')
  })

  it('should render component with own formatter and match snapshot', () => {
    const wrapper = mountFunction({
      propsData: {
        value: '2005-11',
        format: value => `(${value})`,
      },
    })

    expect(wrapper.findAll('.g-date-picker-header__value div').at(0).element.textContent.trim()).toBe('(2005-11)')
  })

  it('should render component with default slot and match snapshot', () => {
    const wrapper = mountFunction({
      propsData: {
        value: '2005-11',
      },
      slots: {
        default: '<span>foo</span>',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should trigger event on selector click', () => {
    const wrapper = mountFunction({
      propsData: {
        value: '2005-11',
      },
    })

    const toggle = jest.fn()
    wrapper.vm.$on('toggle', toggle)

    wrapper.findAll('.g-date-picker-header__value button').at(0).trigger('click')
    expect(toggle).toHaveBeenCalled()
  })
})
