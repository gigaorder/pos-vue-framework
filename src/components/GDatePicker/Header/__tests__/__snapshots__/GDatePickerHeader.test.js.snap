// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GDatePickerHeader.js should render component and match snapshot 1`] = `
<div class="g-date-picker-header">
  <div class="g-date-picker-header__value">
    <div name="tab-transition">
      <button type="button">
        November 2005
      </button>
    </div>
  </div>
  <button class="g-date-picker-header__prev-button">
  </button>
  <button class="g-date-picker-header__next-button">
  </button>
</div>
`;

exports[`GDatePickerHeader.js should render component with default slot and match snapshot 1`] = `
<div class="g-date-picker-header">
  <div class="g-date-picker-header__value">
    <div name="tab-transition">
      <button type="button">
        <span>
          foo
        </span>
      </button>
    </div>
  </div>
  <button class="g-date-picker-header__prev-button">
  </button>
  <button class="g-date-picker-header__next-button">
  </button>
</div>
`;

exports[`GDatePickerHeader.js should render disabled component and match snapshot 1`] = `
<div class="g-date-picker-header g-date-picker-header--disabled">
  <div class="g-date-picker-header__value g-date-picker-header__value--disabled">
    <div name="tab-transition">
      <button type="button">
        November 2005
      </button>
    </div>
  </div>
  <button disabled="disabled"
          class="g-date-picker-header__prev-button"
  >
  </button>
  <button disabled="disabled"
          class="g-date-picker-header__next-button"
  >
  </button>
</div>
`;

exports[`GDatePickerHeader.js should render readonly component and match snapshot 1`] = `
<div class="g-date-picker-header">
  <div class="g-date-picker-header__value">
    <div name="tab-transition">
      <button type="button">
        November 2005
      </button>
    </div>
  </div>
  <button class="g-date-picker-header__prev-button">
  </button>
  <button class="g-date-picker-header__next-button">
  </button>
</div>
`;

exports[`GDatePickerTitle.js should render component and match snapshot 1`] = `
<div class="g-date-picker-header">
  <div class="g-date-picker-header__value">
    <div name="tab-transition">
      <button type="button">
        November 2005
      </button>
    </div>
  </div>
  <button class="g-date-picker-header__prev-button">
  </button>
  <button class="g-date-picker-header__next-button">
  </button>
</div>
`;

exports[`GDatePickerTitle.js should render component with default slot and match snapshot 1`] = `
<div class="g-date-picker-header">
  <div class="g-date-picker-header__value">
    <div name="tab-transition">
      <button type="button">
        <span>
          foo
        </span>
      </button>
    </div>
  </div>
  <button class="g-date-picker-header__prev-button">
  </button>
  <button class="g-date-picker-header__next-button">
  </button>
</div>
`;

exports[`GDatePickerTitle.js should render disabled component and match snapshot 1`] = `
<div class="g-date-picker-header g-date-picker-header--disabled">
  <div class="g-date-picker-header__value g-date-picker-header__value--disabled">
    <div name="tab-transition">
      <button type="button">
        November 2005
      </button>
    </div>
  </div>
  <button disabled="disabled"
          class="g-date-picker-header__prev-button"
  >
  </button>
  <button disabled="disabled"
          class="g-date-picker-header__next-button"
  >
  </button>
</div>
`;

exports[`GDatePickerTitle.js should render readonly component and match snapshot 1`] = `
<div class="g-date-picker-header">
  <div class="g-date-picker-header__value">
    <div name="tab-transition">
      <button type="button">
        November 2005
      </button>
    </div>
  </div>
  <button class="g-date-picker-header__prev-button">
  </button>
  <button class="g-date-picker-header__next-button">
  </button>
</div>
`;
